nessie/src/test/java/org/apache/iceberg/nessie/BaseTestIceberg.java

```diff
@@ -0,0 +1,171 @@
+
+  @BeforeClass
+  public static void create() throws Exception {
+    tempDir = java.nio.file.Files.createTempDirectory(
+        "test",
+        PosixFilePermissions.asFileAttribute(PosixFilePermissions.fromString("rwxrwxrwx")))
+        .toFile();
+  }
+
+  public BaseTestIceberg(String branch) {
+    this.branch = branch;
+  }
+
+  private void resetData() throws NessieConflictException, NessieNotFoundException {
+    for (Reference r : tree.getAllReferences()) {
+      if (r instanceof Branch) {
+        tree.deleteBranch(r.getName(), r.getHash());
+      } else {
+        tree.deleteTag(r.getName(), r.getHash());
+      }
+    }
+    tree.createReference(Branch.of("main", null));
+  }
+
+  @Before
+  public void beforeEach() throws NessieConflictException, NessieNotFoundException {
+    String port = System.getProperty("quarkus.http.test-port", "19120");
+    String path = String.format("http://localhost:%s/api/v1", port);
+    this.client = NessieClient.none(path);
+    tree = client.getTreeApi();
+    contents = client.getContentsApi();
+
+    resetData();
+
+    try {
+      tree.createReference(Branch.of(branch, null));
+    } catch (Exception e) {
+      // ignore, already created. Cant run this in BeforeAll as quarkus hasn't disabled auth
+    }
+
+    hadoopConfig = new Configuration();
+    hadoopConfig.set(NessieClient.CONF_NESSIE_URL, path);
+    hadoopConfig.set(NessieClient.CONF_NESSIE_REF, branch);
+    hadoopConfig.set(NessieClient.CONF_NESSIE_AUTH_TYPE, "NONE");
+    hadoopConfig.set("nessie.warehouse.dir", tempDir.toURI().toString());
+    catalog = initCatalog(branch);
+  }
+
+  NessieCatalog initCatalog(String ref) {
+    NessieCatalog newCatalog = new NessieCatalog();
+    newCatalog.setConf(hadoopConfig);
+    newCatalog.initialize(null, ImmutableMap.of(NessieClient.CONF_NESSIE_REF, ref));
+    return newCatalog;
+  }
+
+  protected Table createTable(TableIdentifier tableIdentifier, int count) {
+    try {
+      return catalog.createTable(tableIdentifier, schema(count));
+    } catch (Throwable t) {
+      LOGGER.error("unable to do create " + tableIdentifier.toString(), t);
+      throw t;
+    }
+  }
+
+  protected void createTable(TableIdentifier tableIdentifier) {
+    Schema schema = new Schema(StructType.of(required(1, "id", LongType.get()))
+                                         .fields());
+    catalog.createTable(tableIdentifier, schema).location();
+  }
+
+  protected static Schema schema(int count) {
+    List<Types.NestedField> fields = new ArrayList<>();
+    for (int i = 0; i < count; i++) {
+      fields.add(required(i, "id" + i, Types.LongType.get()));
+    }
+    return new Schema(Types.StructType.of(fields).fields());
+  }
+
+  void createBranch(String name, String hash) throws NessieNotFoundException, NessieConflictException {
+    if (hash == null) {
+      tree.createReference(Branch.of(name, null));
+    } else {
+      tree.createReference(Branch.of(name, hash));
+    }
```

Comment:
Looks like this could just always pass `hash`.

Reply:
:-) i think you are correct